File: src/main/java/xyz/terrific/server/client/cnc/ClientServer.java
package xyz.terrific.server.client.cnc;


import java.io.IOException;
import java.net.ServerSocket;
import java.net.Socket;

public class ClientServer {
    private final ServerSocket socket;

    public ClientServer(ServerSocket socket) {
        this.socket = socket;
    }

    public static void init() throws IOException {

        ServerSocket cSocket    = new ServerSocket(1773);
        ClientServer server     = new ClientServer(cSocket);
        server.start();

    }

    public void start() {
        try {
            while (!socket.isClosed()) {

                Socket conn = socket.accept();

                System.out.printf(" [ CLIENT | %s ] Connected\n", conn.getRemoteSocketAddress().toString());
                ClientHandler handler = new ClientHandler(conn);

                Thread thread = new Thread(handler);
                thread.start();
            }

        } catch (Exception e) {
            e.printStackTrace();
        }

    }

    public void close() {

        try {
            if (socket != null)
                socket.close();
        } catch (Exception e) {
            e.printStackTrace();
        }

    }

}

File: src/main/java/xyz/terrific/server/client/cnc/ClientServer.kt
package xyz.terrific.server.client.cnc

import java.io.IOException
import java.net.ServerSocket

/**
 * @author TerrificTable55
 * @version 1.0
 */
class ClientServer
    /**
     * @param socket sets private socket variable
     */
    (private val socket: ServerSocket?) {

    /**
     * @use start function
     */
    fun start() {
        try {
            while (!socket!!.isClosed) {
                val conn = socket.accept()
                System.out.printf(" [ CLIENT | %s ] Connected\n", conn.remoteSocketAddress.toString())
                val handler = ClientHandler(conn)
                val thread = Thread(handler)
                thread.start()
            }
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    /**
     * @use closes socket
     */
    fun close() {
        try {
            socket?.close()
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    companion object {
        @Throws(IOException::class)
        fun init() {
            val cSocket = ServerSocket(1773)
            val server = ClientServer(cSocket)
            server.start()
        }
    }
}
